### PR TITLE
Create UUID when system_uuid file empty

### PR DESCRIPTION
**Pull request by s-leon**
Added a check to only read the uuid-file if it is not empty, otherwise it creates a new uuid.
(... and added a dependency to readme)

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> in the <code>package</code> directory to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/5943/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 250d17032e8d8e7b2ded36295c53123857b7a5c1 into 6ee70a806ae43e81e8c77ae2462435dd48f879c7)

File: lib/suse/connect/api.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def make_paginated_request(method, url, options = {})
       private
 
       def system_uuid
-        @system_uuid ||= if File.exist?(UUID_FILE_LOCATION)
+        @system_uuid ||= if File.exist?(UUID_FILE_LOCATION) && !File.empty?(UUID_FILE_LOCATION)
```
Comment (fschueller) [inline lib/suse/connect/api.rb]: Maybe add or adapt the test for this line in `spec/suse/connect/api.spec`.

Comment (s-leon) [reply lib/suse/connect/api.rb]: _done!_ I've added a test, that creates an empty file and checks if the "test-uuid" is present. (like it is done with the existing tests)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/7400/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 5274419bb46e60dffa7ea9edc690124b2aca6de0 into 6ee70a806ae43e81e8c77ae2462435dd48f879c7)

File: spec/suse/connect/api_spec.rb [hunk 2]
```diff
@@ -29,6 +30,18 @@
         expect(method_call).to be(uuid)
       end
     end
+
+    context 'when system_uuid file is empty' do
```
[the remark this answers is not in the document]
Comment (s-leon) [reply spec/suse/connect/api_spec.rb]: Isn't the fakefs route more robust to code changes? Is there a drawback to using it?
As for code style, i can change that. :smiley:

[the remark this answers is not in the document]
Comment (fschueller) [reply spec/suse/connect/api_spec.rb]: Technically the `FakeFS` was probably ok, it just looked more than an integration/feature than a unit test to me :) However the file path was hardcoded in the test - in that regard it's not that robust.

My suggestion is to follow the style of testing for the previous test examples. If you stick to mocking the file, I'd add expectations for the return values of `exist?` and `empty?` - however, IMO it's not that clean to put multiple expects in one `it` block.

**Pull request merged**